### PR TITLE
feat: changing to setuptools_scm

### PR DESCRIPTION
**Pull request by nilodna**
Moving from versioneer to setuptools_scm. 

I'm not sure if everything was covered, but I'm counting on @ocefpaf's eyes to check it :P

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,46 +1,11 @@
-from pathlib import Path
-
-from setuptools import find_packages, setup
-
-import versioneer
```
Comment (ocefpaf) [inline setup.py:5]: This line should go but the rest must be re-add or, we need a pyproject.toml with these info. I would not do that in the same PR just to keep it a bit simpler for now.

File: setup.py [hunk 2]
```diff
@@ -1,46 +1,11 @@
...
-        "Environment :: Console",
-        "Intended Audience :: Science/Research",
-        "Operating System :: OS Independent",
-        "Topic :: Scientific/Engineering",
-    ],
-    platforms="any",
-    packages=find_packages(),
-    extras_require={"testing": ["pytest"]},
-    install_requires=install_requires,
-    cmdclass=versioneer.get_cmdclass(),
```
Comment (ocefpaf) [inline setup.py:44]: Remove this one too.

File: setup.py [hunk 3]
```diff
@@ -1,46 +1,11 @@
...
-    entry_points={"console_scripts": ["gridio = gridgeo.gridio:main"]},
+    use_scm_version={
+        "write_to": "gridgeo/_version.py",
+        "write_to_template": '__version__ = "{version}"',
+        "tag_regex": r"^(?P<prefix>v)?(?P<version>[^\+]+)(?P<suffix>.*)?$",
+    },
+    setup_requires=['setuptools_scm'],
```
Comment (ocefpaf) [inline setup.py:10]: These are fine and, with the rest restore, the tests should pass.

**Comment by nilodna** [comment]
Hi @ocefpaf, I'm not sure if I got all your suggestions right. I applied all of them, but the tests keep failing. Did I miss something?

**Comment by ocefpaf** [comment]
> Hi @ocefpaf, I'm not sure if I got all your suggestions right. I applied all of them, but the tests keep failing. Did I miss something?
> 
> The test log is pointing to an error when loading the `get_versions()` function because there is no ` _version.py`. Do you have any clue about what happened?

I forgot about the `__init__.py`

You need to change:

```python
from ._version import get_versions

__version__ = get_versions()["version"]
del get_versions
```

to 

```python
try:
    from ._version import __version__
except ImportError:
    __version__ = "unknown"
```

**Comment by ocefpaf** [comment]
One more place I forgot:

Change the version in https://github.com/pyoceans/gridgeo/blob/main/docs/source/conf.py#L58-L65

to

```python
import gridgeo

version = release = gridgeo.__version__
```

**Pull request merged**